lib/absinthe/phase/schema/debugger.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Absinthe.Phase.Schema.Debugger do
+  @moduledoc false
+
+  use Absinthe.Phase
+  alias Absinthe.Blueprint
+
+  def run(blueprint, _opts) do
+    blueprint = Blueprint.prewalk(blueprint, &inject_name(&1, blueprint))
```

Comment:
In order to render SDL from the individual Blueprint structs (w/o a reference to the entire blueprint), I had to walk the blueprint and inject names 

* `interfaces` referenced in objects
* type names referenced by `non_null` and `list` references

To get this working quickly, I just extended the blueprint structs, but I think I'd rather have these generate the same structure that is generated when an SDL based schema is turned into blueprints.

Reply:
@bruce Do you know where this difference originates? and more importantly, the right place to normalize so they all contain a `TypeReference.Name`?

```elixir
# Macro:
%Absinthe.Blueprint.TypeReference.NonNull{
  errors: [],
  of_type: %Absinthe.Blueprint.TypeReference.List{
    errors: [],
    of_type: %Absinthe.Blueprint.TypeReference.NonNull{
      errors: [],
      of_type: :category,
    },
  },
}

# SDL:
%Absinthe.Blueprint.TypeReference.NonNull{
  errors: [],
  of_type: %Absinthe.Blueprint.TypeReference.List{
    errors: [],
    of_type: %Absinthe.Blueprint.TypeReference.NonNull{
      errors: [],
      of_type: %Absinthe.Blueprint.TypeReference.Name{
        errors: [],
        name: "Category",
        schema_node: nil
      },
    },
  },
}
```

Reply:
I think you should probably normalize in `Absinthe.Phase.Schema.NormalizeReferences`

Reply:
`NormalizeReferences` says it's a hack that will be removed...

https://github.com/absinthe-graphql/absinthe/blob/master/lib/absinthe/phase/schema/normalize_references.ex#L4

Also, it seems like it's doing the reverse of what I want to do here :) (atom -> reference)

It seems like a problem if I need to run a `Blueprint` through phases if we want it to be `inspect`-able, since we won't have the phases or the schema